### PR TITLE
fix write lock error in live backup test

### PR DESCRIPTION
**Pull request by xutian**
Enable force_share flag to avoid shared write lock issue. Pause VM tto ensure image meta not change by running VM. And, destory VM before do compare image step to ensure image not change by VM during the comparison process.

Signed-off-by: XuTian <xutian@redhat.com>

ID: 1628432

### CODE REVIEW

**Comment by xutian** [comment]
Local test pass, see result below:
```
[root@hp-dl385g10-21 scripts]# ./scan_results.py 
CaseName                                 Status          StartTime       EndTime         Result     TimeElapsed
'1-Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.live_backup' Finshed 2019-06-12 22:14:09 2019-06-12 22:22:09 PASS 480.34943
Summary: 
Finshed=1, PASS=1```

**Comment by xutian** [comment]
@aliang123  @ngu-niny  would you help to review this request, thanks.

**Comment by aliang123** [comment]
(1/6) Host_RHEL.m7.u7.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.7.x86_64.io-github-autotest-qemu.live_backup: PASS (1082.71 s)
 (2/6) Host_RHEL.m7.u7.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.7.x86_64.io-github-autotest-qemu.live_backup.with_data_plane: PASS (1062.78 s)
 (3/6) Host_RHEL.m7.u7.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.7.x86_64.io-github-autotest-qemu.live_backup.granularity.32768: PASS (957.39 s)
 (4/6) Host_RHEL.m7.u7.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.7.x86_64.io-github-autotest-qemu.live_backup.granularity.32768.with_data_plane: PASS (977.87 s)
 (5/6) Host_RHEL.m7.u7.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.7.x86_64.io-github-autotest-qemu.live_backup.granularity.131072: PASS (966.59 s)
 (6/6) Host_RHEL.m7.u7.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.7.x86_64.io-github-autotest-qemu.live_backup.granularity.131072.with_data_plane: PASS (978.90 s)

**Comment by aliang123** [comment]
(1/3) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.live_backup: PASS (260.63 s)
 (2/3) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.live_backup.granularity.32768: PASS (191.11 s)
 (3/3) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.live_backup.granularity.131072: PASS (257.09 s)

Acked-by: Aihua Liang

**Comment by ngu-niny** [comment]
Test PASS:
 
(1/3) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.live_backup: PASS (399.26 s)
 (2/3) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.live_backup: PASS (389.50 s)
 (3/3) Host_RHEL.m8.u1.qcow2.spapr_vscsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.live_backup: PASS (760.94 s)

Acked-by: Nini Gu <ngu@redhat.com>

File: qemu/tests/live_backup_base.py [hunk 1]
```diff
@@ -145,9 +149,20 @@ def get_image_size(self, image):
...
         :return: image size.
         """
+        force_share = False
+        pause_vm = False
         params = self.params.object_params(image)
         qemu_image = qemu_storage.QemuImg(params, self.data_dir, image)
-        image_info = qemu_image.info()
+        if self.vm:
+            pids = process.getoutput(
+                "lsof %s |grep -v PID|awk '{print $2}'" % qemu_image.image_filename)
```
Comment (lolyu) [inline qemu/tests/live_backup_base.py]: better to simply use `"lsof -t %s"`

Comment (xutian) [reply qemu/tests/live_backup_base.py]: @lolyu  thanks your comments,  "-t" options  is not always supported.

**Pull request merged**